### DISCUSSION TITLE
Technologien

### DISCUSSION THREAD

**Discussion by codajoda** [SE-TINF22B2/G3-ApeRepublic#3] [Ideas]
Technologien

**Comment by codajoda** [comment]
# 🔧 Mögliche Technologien

## Frontend
* Flutter (Zuständig: @Marc738)  ❌ 
* React  ❌ 
* Angular (Zuständig: @codajoda)  ✅ 
* SCSS (Sassy CSS)  ✅ 
* Tailwind  ❌ 
* HTML, CSS & JS  ❌ 

## Backend
* Node.js ❌ 
* Deno ✅ 
* Bun ❌ (Nicht Windows-kompartibel)

## Datenbank
* MongoDB  ✅ 
* MySQL  ❌ 

## Hosting
* GitHub -> https://pages.github.com/ (Zuständig: @Marc738 )
* DHBW Cloud (Zuständig: -)
* Webserver (beliebig)

**Reply by elnaz-gharoon** [reply]
_Frontend: Flutter oder Kotlin?_ 


Kotlin: 

**PRO** 👍 

- Unbekannt für die meisten im Team ABER eine sehr wichtige Eigenschaft von Kotlin ist eine Interoperabilität mit Java, was es für Entwickler relativ einfach macht, Java zu erlernen und auf Kotlin umzusteigen. 


- NullPointerException 
In den frühen Phasen der Entwicklung können App-Entwickler die Vorteilen der Vorteile der NullPointerException Funktion nutzen

- Zukunftssicherheit: 

Kotlin ist eine stabilere und etabliertere Technologie, insbesondere in der Android-Entwicklung, und könnte daher eine sicherere Wahl für Projekte sein, die eine lange Lebensdauer haben sollen.

**Con** 👎 

-Plattformübergreifende Entwicklung:
Mit Kotlin Multiplatform Mobile (KMM) bietet Kotlin auch eine plattformübergreifende Lösung, aber es ist nicht so ausgereift wie Flutter in Bezug auf plattformübergreifende Funktionen.


- Wenn Application darin besteht, Daten von einer APIs, (Zu diesen APIs könnten Endpunkte gehören, um Marktdaten/Watchlist, Preise, Benutzerkontoinformationen,) abzurufen und anzuzeigen, , ohne eine intensive Interaktion mit der Hardware zu erfordern, ist Flutter eine geeignete Wahl. In solchen Fällen, in denen die Hauptaufgabe darin besteht, Daten anzuzeigen, kann Flutter effizient sein.

https://www.digitalocean.com/community/tutorials/flutter-flutter-http

**Reply by elnaz-gharoon** [reply]
Backend : MongoDB vs. SQLit (often used in mobile Apps) ?

**Comment by Marc738** [comment]
# Flutter Web
## 🏃 Pro
* Code deploy-bar alls App auf Mac/Windows/iOS/Android
* Große Auswahl an Widgets
* Basiert auf Dart, welches Java sehr ähnlich ist
* Leichte Umsetzung von API-Calls

## 🌵 Neutral
* VSC oder Android Studio und Chrome für Entwicklung

## 🤒 Con
* Unbekannt für die meisten im Team
* Nicht geeignet für Suchmaschinen -> Code wird dynamisch generiert und liegt nicht in einer .html-Datei.
* Scroll-Verhalten komisch (Stand 2021)
* Welche Widgets sind verwendbar -> Nicht jedes Widget ist automatisch webkompartibel.

**Comment by Marc738** [comment]
# 🥇 Angular
- Angular wird mit TypeScript geschrieben, aber da JavaScript auch TypeScript ist, sollte das kein Problem sein.
- dynamisches HTML möglich (mit if, for-Schleifen und sowas...)
- Components (mit eigenen Variablen, Methoden, und html-Template) können gebaut werden -> Übersichtlichkeit der html-Struktur, da nun Components wiederverwendet werden können und ein einheitlicher Look der Anwendung entsteht
- VSC besitzt useful Tools zu Angular (bauen von Components...)
- Mit Angular können Web-Anwendungen auch als Native Anwendungen benutzt werden 
- Angular project kann mitsamt Struktur durch cmd direkt erstellt werden 
- Viele Libraries zu Material Design z.B. Nebular, theoretisch muss man dann keine Gedanken über Design mehr machen
- Angular beinhaltet Test-Dateien für Unit-Tests und son shit
- Es gibt genug Tutorials zu Angular (Angular wird von Google entwickelt)
